cocos/core/event/async-delegate.ts

```diff
@@ -104,6 +104,11 @@ export class AsyncDelegate<T extends (...args: any) => (Promise<void> | void) =
      * @returns @en The promise awaiting all async callback resolved. @zh 等待所有异步回调结束的 Promise 对象。
      */
     public dispatch (...args: Parameters<T>) {
-        return Promise.all(this._delegates.map((func) => func(...arguments)).filter(Boolean));
+        if(window.oh) {
```

Comment:
How about using a macro to clip the implementation which likes:
```ts
if (JSB) {...}
or 
if (NATIVE) {...}
```
probably, needs a new macro likes
```ts
if (OPENHARMONY) {...}
```

Reply:
This is a bug, it will be fixed later

Reply:
OK, so it's better to add a TODO comment.